lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python
+# (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# most of it copied from AWX's scan_packages module
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: package_facts
+short_description: package information as facts
+description:
+  - Return information about installed packages as facts
+options:
+  manager:
+    description:
+      - The package manager used by the system so we can query the package information
+    default: autodetect
+    choices: ["auto", "rpm", "zypper", "dnf", "yum", "dpkg", "apt", "aptitude"]
+    required: False
+version_added: "2.5"
+author:
+  - Matthew Jones
+  - Brian Coca
+  - Adam Miller
+'''
+
+EXAMPLES = '''
+- name: get the rpm package facts
+  package_facts:
+    manager: "rpm"
+
+- name: show them
+  debug: var=ansible_facts.packages.rpm
+
+
+
+- name: get the deb package facts
+  package_facts:
+    manager: "apt"
+
+- name: show them
+  debug: var=ansible_facts.packages.rpm
+'''
+
+RETURN = '''
+ansible_facts:
+  description: facts to add to ansible_facts
+  returned: always
+  type: complex
+  contains:
+    packages.rpm:
+      description: list of dicts with package information
+      returned: when rpm, yum, dnf, or zypper is the specified or auto detected manager
+      type: dict
+      sample:
+        {
+          "yum": [
+            {
+                "arch": "noarch",
+                "epoch": null,
+                "name": "yum",
+                "release": "513.fc27",
+                "source": "rpm",
+                "version": "3.4.3"
+            }
+          ],
+          "yum-utils": [
```

Comment:
I would include an example of a package that is installed multiple times ('kernel' for ex) to demonstrate how that is represented.

Reply:
:+1: Will do.